tracer/contrib/net/httptrace/httptrace.go

```diff
@@ -0,0 +1,122 @@
+package httptrace
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// HttpTracer is used to trace requests in a net/http server.
+type HttpTracer struct {
+	tracer  *tracer.Tracer
+	service string
+}
+
+// NewHttpTracer creates a HttpTracer for the given service and tracer.
+func NewHttpTracer(service string, t *tracer.Tracer) *HttpTracer {
+	t.SetServiceInfo(service, "net/http", ext.AppTypeWeb)
+	return &HttpTracer{
+		tracer:  t,
+		service: service,
+	}
+}
+
+// Handler will return a Handler that will wrap tracing around the
+// given handler.
+func (h *HttpTracer) Handler(handler http.Handler) http.Handler {
+	return h.TraceHandlerFunc(http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+		handler.ServeHTTP(writer, req)
+	}))
+}
+
+// TraceHandlerFunc will return a HandlerFunc that will wrap tracing around the
+// given handler func.
+func (h *HttpTracer) TraceHandlerFunc(handler http.HandlerFunc) http.HandlerFunc {
+
+	return func(writer http.ResponseWriter, req *http.Request) {
+
+		// bail out if tracing isn't enabled.
+		if !h.tracer.Enabled() {
+			handler(writer, req)
+			return
+		}
+
+		// trace the request
+		tracedRequest, span := h.trace(req)
+		defer span.Finish()
+
+		// trace the response
+		tracedWriter := newTracedResponseWriter(span, writer)
+
+		// run the request
+		handler(tracedWriter, tracedRequest)
+	}
+}
+
+// span will create a span for the given request.
+func (h *HttpTracer) trace(req *http.Request) (*http.Request, *tracer.Span) {
+	resource := req.Method + " " + req.URL.Path
```

Comment:
Hello, thank you for that PR FreekingDean. I actually found a nice pattern matching function in `net/url`. You can try this on the playground:
```
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u, _ := url.Parse("/books?author=:author&subject=")
	fmt.Println(u.Path)
	u, _ = url.Parse("/search?q=net%2Furl&rlz=1C5CHFA_enUS730US731&oq=net%2Furl&aqs=chrome..69i57j69i58j69i61l3j69i59.1391j0j7&sourceid=chrome&ie=UTF-8#q=golang+parse+url+path")
	fmt.Println(u.Path)
}
```